src/Domain/ValueObjects/WalletValueObject.cs

```diff
@@ -0,0 +1,19 @@
+using Domain.Aggregates;
+namespace Domain.ValueObjects;
+
+public record Wallet(decimal Balance){
```

Comment:
this is a very nice improvement. please read [C# record docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record) for more info on records